Navigation/Profile/ProfileHeaderView.swift

```diff
@@ -0,0 +1,143 @@
+        button.setTitleColor(.white, for: .normal)
+        button.layer.cornerRadius = 12
+        button.layer.shadowColor = UIColor.black.cgColor     // cgColor for layer.
+        button.layer.shadowOffset = CGSize(width: 4, height: 4)
+        button.layer.shadowOpacity = 0.7
+        button.layer.shadowRadius = 4
+        button.backgroundColor = .systemBlue
+    // Instead IBAction.
+        button.addTarget(self, action: #selector(buttonPressed), for: .touchUpInside)
+        return button
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        setUpProfileViews()
+        
+    }
+    // Calling for storyboards.
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        
+        setUpProfileViews()
+        
+    }
+    
+    @objc private func buttonPressed() {
+        statusLabel.text = (statusText ?? "").isEmpty ? "Waiting for something..." : statusText
+    }
+    
+    //  ( _ textField: UITextField) - RESPONDER
+    @objc private func statusTextChanged(_ textField: UITextField) {
+        statusText = textField.text
+    }
+    
+    private func setUpProfileViews() {
+        [avatarImageView,
+         nameLabel,
+         statusLabel,
+         statusTextField,
+         button].forEach {
+            // Enables autolayout for our imageView.
+            $0.translatesAutoresizingMaskIntoConstraints = false
+            self.addSubview($0)
+        }
+        
+        NSLayoutConstraint.activate([
+            avatarImageView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 16),
+            avatarImageView.leftAnchor.constraint(equalTo: safeAreaLayoutGuide.leftAnchor, constant: 16),
+            avatarImageView.widthAnchor.constraint(equalToConstant: avatarWidth),
+            avatarImageView.heightAnchor.constraint(equalToConstant: avatarWidth),
+            
+            nameLabel.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 27),
+            nameLabel.leftAnchor.constraint(equalTo: safeAreaLayoutGuide.leftAnchor, constant: 160),
+            nameLabel.widthAnchor.constraint(equalTo: safeAreaLayoutGuide.widthAnchor, constant: -180),
+            
+            statusLabel.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 80),
+            statusLabel.leftAnchor.constraint(equalTo: safeAreaLayoutGuide.leftAnchor, constant: 160),
+            statusLabel.widthAnchor.constraint(equalTo: safeAreaLayoutGuide.widthAnchor, constant: -180),
+            
+            statusTextField.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 110),
+            statusTextField.leftAnchor.constraint(equalTo: safeAreaLayoutGuide.leftAnchor, constant: 160),
+            statusTextField.widthAnchor.constraint(equalTo: safeAreaLayoutGuide.widthAnchor, constant: -180),
+            statusTextField.heightAnchor.constraint(equalToConstant: 40),
```

Comment:
<img width="369" alt="image" src="https://user-images.githubusercontent.com/67570073/149528071-383bf484-55f5-4d4e-893f-50c0d938f09e.png">
for nameLabel, statusLabel and statusTF you can take leading constraint equals to avatar trailing plus 16